.cursor/rules/ux-standards.mdc

```diff
@@ -0,0 +1,238 @@
+- Always require user confirmation (e.g., a modal) for destructive actions to prevent accidental data loss
+- Make destructive actions visually distinct, using the "negative" color palette to indicate danger or permanence
+- Place destructive actions away from primary actions to avoid accidental clicks
+- Provide an option to undo the action where feasible (e.g., allowing users to restore deleted items within a certain timeframe)
+
+### External Links & Documentation Links
+- For all external links or links to documentation, use `target="_blank"` to open in a new tab
+- Always include `rel="noopener noreferrer"` for security and performance best practices when opening links in a new tab (e.g., `<a href="https://docs.example.com" target="_blank" rel="noopener noreferrer">View Documentation</a>`)
+
+## User Guidance & Feedback Mechanisms
+
+### Error Messages (Dialogs, In-Line, Full Pages)
+- Use plain, empathetic language that explains what went wrong without sounding accusatory (see "Error Messaging" section for language guidelines)
+- Importantly, explain what actions the user can take to remedy the issue, if any
+- Display error messages close to the element causing the issue (e.g., inline validation errors for form fields)
+- Provide a link to documentation that explains how to work with the desired entity (e.g. "See docs.")
+- Ensure that doc links point to the correct Community or Enterprise URL (using an existing utility if the link would be available on both)
+- Offer a "Retry" or "Refresh" button if that is possible
+- Avoid displaying stack trace messages by default; allow the user to dig deeper if necessary by clicking on a "View error details" button
+- Clear error messages automatically when the error no longer applies
+
+### Empty States
+- Offer clear guidance on how to populate the state (e.g., "Start by importing your dataset!")
+- Use inviting illustrations or friendly language
+- Include links to relevant documentation
+
+### Tooltips
+- Use tooltips to provide context or explain the functionality of less obvious features without cluttering the interface
+- Provide concise tooltips for additional context on hover without overwhelming the main interface, especially on icon-only buttons
+- For disabled features, provide explanations via tooltips (e.g., "This option is available when Task assignment is automatic")
+- Ensure tooltips are concise, helpful, and disappear when users move away from the element
+- Make tooltips keyboard accessible and announced by screen readers using appropriate ARIA attributes
+- Use `aria-label` combined with tooltips for comprehensive accessibility on icon-only buttons (e.g., icon-only button with `aria-label="Delete project"` and tooltip showing "Permanently delete this project")
+- Avoid relying solely on tooltips for critical information, as they may be missed by some users
+
+
+### Loading States, Progress, and Recovery
```

Comment:
I would add advice to avoid jumping states when user might click on a wrong place because layout shifted right after the loading is finished